org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/regression/UnnamedModuleTest.java

```diff
@@ -66,4 +71,35 @@ public void testBug522326() {
 		}
 	);
 }
+
+// https://github.com/eclipse-jdt/eclipse.jdt.core/issues/415
+// Should compile successfully when the compilation unit contains a split package and
+// the option 'OPTION_IgnoreUnnamedModuleForSplitPackage' is turned on.
+public void testIgnoreUnnamedModule() {
+	String path = this.getCompilerTestsPluginDirectoryPath() + File.separator + "workspace" + File.separator + "ignore-unnamed-module-test.jar";
+	String[] defaultLibs = getDefaultClassPaths();
+	int len = defaultLibs.length;
+	String[] libs = new String[len+1];
+	System.arraycopy(defaultLibs, 0, libs, 0, len);
+	libs[len] = path;
+	Map<String, String> compilerOptions = getCompilerOptions();
+	compilerOptions.put(CompilerOptions.OPTION_IgnoreUnnamedModuleForSplitPackage, CompilerOptions.ENABLED);
+	this.runConformTest(
+			true,
+			new String[] {
+				"X.java",
+				"import org.xml.sax.SAXParseException;\n" +
```

Comment:
This successfully tests the code path originating from `CompilationUnitScope.faultInImports()`.

Additionally, I'd like to see one test covering the case where this is triggered from `QualifiedTypeReference.getTypeBinding(Scope)`.

Currently, we know that `ProblemReporter.conflictingPackagesFromModules()` can only be triggered by a specific outcome from `SplitPackageBinding.getVisibleFor()`, good. But from a testing p.o.v. it might be prudent to take a more black-boxish stand, i.e., test all scenarios that can pull in a package into complation - including qualified type references.

Bonus questions: what happens when code uses a `QualifiedNameReference` pointing directly to a static member of an affected type? Do we actually detect the error? If so, is the new option effective in that case, too?

Reply:
Thanks for the suggestion, more cases are added in the latest commit.

In terms of the bonus question, before this change:

If the code is written as:

```java
import static a.split.package.type.staticMember;
```
JDT will report the split package error.

But if the fully qualified name is directly used in the code:

```java
String s = a.split.package.type.staticMember;
```

JDT will not report any errors.

But anyway, I put both of the 2 cases into the tests to avoid regression in the future.